# Individual Plan

#### Github link: https://github.com/Coreeper/DSCI-Project-17

In [1]:
library(tidyverse)
library(repr)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


## 1. Importing Data

In [2]:
url_player = "https://drive.google.com/uc?export=download&id=1EDNOF0c9WQe5BaJwc-dlqKKLPObhGniP"
url_session = "https://drive.google.com/uc?export=download&id=1zNZviQwtpxZs5HitaQkeaSJ5QjPerdZ-"

In [3]:
player = read_csv (url_player)
session = read_csv (url_session)
head(player)
head(session)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

#### 2. Data Analyze For `player.csv`

In [4]:
stat_played_hour <- player |> 
summarise(min(played_hours), max(played_hours), mean(played_hours))

stat_age <- player |> 
summarise(min(Age,na.rm=TRUE), max(Age,na.rm=TRUE), mean(Age,na.rm=TRUE))

stat_age
stat_played_hour

min(Age, na.rm = TRUE) max(Age, na.rm = TRUE) mean(Age, na.rm = TRUE)
1 9                      58                     21.13918

min(played_hours) max(played_hours) mean(played_hours)
1 0                 223.1             5.845918

For `player.csv` , there are 196 observations and 7 columns.
1. The column `experience` is charactor type that describes the level of players as one of: `Pro`, `Veteran`, `Amateur`, `Regular`, and `Beginner`.
2. The column `subscribe` is logical type that describes whether the player subscribed to the game info as `TRUE` and `FALSE`.
3. The column `hashedEmail` is charactor type that describes hashing encoded email address of the player.
4. The column `played_hours` is double type that describes the player's playtime on the server, ranging from 0 to 223.1, with a mean of 5.85.
5. The column `name` is charater type that describes the name of the player.
6. The column `gender` is charater type that describes the gender of the player as `male` and `female`.
7. The column `Age` is double type that describes the player's age on the server, ranging from 9 to 58, with a mean of 21.14.


#### 3. Data Analyze For `session.csv`

In [5]:
stat_original_start_time <- session |> 
summarise(min(original_start_time), max(original_start_time), mean(original_start_time))

stat_original_end_time <- session |> 
summarise(min(original_end_time, na.rm=TRUE), max(original_end_time, na.rm=TRUE), mean(original_end_time, na.rm=TRUE))

stat_original_start_time
stat_original_end_time

min(original_start_time) max(original_start_time) mean(original_start_time)
1 1.7124e+12               1.72733e+12              1.719201e+12

min(original_end_time, na.rm = TRUE) max(original_end_time, na.rm = TRUE)
1 1.7124e+12                           1.72734e+12                         
  mean(original_end_time, na.rm = TRUE)
1 1.719196e+12

For `session.csv` , there are 1535 observations and 5 columns.
1. The column `hashedEmail` is charactor type that describes hashed email address of the player.
2. The column `start_time` is charactor type that describes the time when the player starts to play on the server.
3. The column `end_time` is charactor type that describes the time when the player leave the server.
4. The column `original_start_time` is double type that uses the millisecond-precision Unix timestamps to describe the time when the player starts to play on the server, ranging from 1.7124e+12 to 1.72733e+12, with a mean of 1.719201e+12.
5. The column `original_end_time` is double type that uses the millisecond-precision Unix timestamps to describe the time when the player leave the server, ranging from 1.7124e+12 to 1.72734e+12, with a mean of 1.719196e+12.